angular-ngrx-scss/src/app/auth/services/auth.service.spec.ts

```diff
@@ -0,0 +1,93 @@
+import {
+  HttpClientTestingModule,
+  HttpTestingController,
+} from '@angular/common/http/testing';
+import { TestBed } from '@angular/core/testing';
+import { environment } from 'src/environments/environment';
+import { SignInResponse, AuthResponse } from '../interfaces/auth';
+import { AuthService } from './auth.service';
+import { TokenService } from './token.service';
+
+describe('AuthService', () => {
+  let tokenService: TokenService;
+  let authService: AuthService;
+  let httpController: HttpTestingController;
+  let token: string | null = 'hello';
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [HttpClientTestingModule],
+      providers: [
+        {
+          provide: TokenService,
+          useValue: {
+            getToken() {
+              return token;
+            },
+            saveToken(token: string) {
```

Comment:
you can omit the param if it's not used
```suggestion
            saveToken() {
```

Reply:
The param for `saveToken` is used to store it in local storage. :)